### ISSUE TITLE
jira init "Unable to generate configuration: EOF"

### ISSUE THREAD

**Issue by schdief** [ankitpokhrel/jira-cli#699]
**Describe the bug**

a few days ago this problem has arised, when using jira cli inside our jenkins pipeline
```
15:15:27  + JIRA_AUTH_TYPE=bearer jira init --force --installation Local --server https://***.***.de/jira1 --login tfcd_bot --project SDM --board 'Lieferhelden Board'
15:15:27  7[?25l8[0G[2K[0;1;92m? [0m[0;1;99mAuthentication type:[0m  [0;36m[Use arrows to move, type to filter, ? for more help][0m
15:15:27  [0;1;36m> basic[0m
15:15:27  [0;39m  bearer[0m
15:15:27  [0;39m  mtls[0m
15:15:27  7[1A[0G[1A[0G[1A[0G8[?25h
15:15:27  [0;31m✗[0m Unable to generate configuration: EOF
```

when I switch from latest image tag to "v1.4.0" it still works, so something broke with v1.5.0:

```
15:17:24  + JIRA_AUTH_TYPE=bearer jira init --force --installation Local --server https://***.***.de/jira1 --login tfcd_bot --project SDM --board 'Lieferhelden Board'
15:17:25  
15:17:25  [0;32m✓[0m Configuration generated: /var/lib/jenkins/workspace/SDM_SockShop_carts_master/.config/.jira/.config.yml
```

I assume somehow the path is changed where the config is stored to and the jenkins pipeline has no access to that path, please instruct how to set the path or restore behaviour from v1.4.0

**Please provide following details**
1. JiraCLI Version:
    ```
    (Version="v1.5.0", GitCommit="8449ab0c27b60caf7ca8bc5fdd5fff876e3672f5", CommitDate="2024-01-09T20:30:43+00:00", GoVersion="go1.21.6", Compiler="gc", Platform="linux/amd64")
    ```
2. Are you using Jira cloud or on-premise jira server?
    ```
    on-premise 9.4.14
    ```
3. What operating system are you using? Also mention version.
    ```
    https://github.com/ankitpokhrel/jira-cli/pkgs/container/jira-cli on Openshift (RHEL)
    ```
4. What terminal are you using? Also mention version.
    ```
    none, Jenkins pipeline "sh"-step
    ```

**To Reproduce**

1. probably hard for you due to our specific setup with Openshift and Jenkins, but you could try running your image as a docker agent within jenkins and probably see the same error

**Expected behavior**
config can be written as with v1.4.0

**Screenshots**
see log output at the beginning

**Additional context**
-

**Comment by ankitpokhrel** [comment]
Hi @schdief, thank you for reporting the issue. The configuration path remains unchanged; however, the problem is because of the new step that asks you to select `Authentication type`. It fails on a non-interactive shell.

I will make a new release with the fix asap.

**Comment by ankitpokhrel** [comment]
Hi @schdief, this should be fixed with #701 and will be available in the next release. The PR adds a new option `--auth-type` that you can use to specify authentication type and skip the prompt. If you already have `JIRA_AUTH_TYPE` env set to `basic`, you don't have to do anything else.

If you want to test the change, you can do so by installing the main branch.

```
# Docker
$ docker run -it --rm ghcr.io/ankitpokhrel/jira-cli:main

# Golang
$ go install github.com/ankitpokhrel/jira-cli/cmd/jira@main

# Homebrew
$ brew tap ankitpokhrel/jira-cli
$ brew install jira-cli --HEAD
```

**Comment by ankitpokhrel** [comment]
Hi @schdief, thanks for testing. I will try to make a new release this week/weekend.
